src/include/daos/mem.h

```diff
@@ -223,13 +261,81 @@ struct umem_instance {
 	umem_class_id_t		 umm_id;
 	const char		*umm_name;
 	PMEMobjpool		*umm_pool;
+	/** Cache the pool id field for umem addresses */
+	uint64_t		 umm_pool_uuid_lo;
+	/** Cache the base address of the pool */
+	uint64_t		 umm_base;
 	/** class member functions */
 	umem_ops_t		*umm_ops;
 };
 
 int  umem_class_init(struct umem_attr *uma, struct umem_instance *umm);
 void umem_attr_get(struct umem_instance *umm, struct umem_attr *uma);
-uint64_t umem_get_uuid(struct umem_instance *umm);
+
+/** Convert an offset to an id.   No invalid flags will be maintained
+ *  in the conversion.
+ *
+ *  \param	umm[IN]		The umem pool instance
+ *  \param	umoff[in]	The offset to convert
+ *
+ *  Returns the mmid
+ */
+static inline umem_id_t
+umem_off2id(const struct umem_instance *umm, umem_off_t umoff)
+{
+	umem_id_t	ummid;
+
+	if (UMOFF_IS_NULL(umoff)) {
+		ummid = UMMID_NULL;
```

Comment:
or just ` return UMMID_NULL;`

Reply:
Ok, can change.  Can it be deferred...Working on btree patch

Reply:
nevermind, went ahead and changed it